lib/node/http.js

```diff
@@ -383,6 +385,108 @@ class HTTP extends Server {
 
       res.json(200, { success: true });
     });
+
+    this.get('/info/name/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      if (!name || !rules.verifyName(name))
+        throw new Error('Invalid parameter.');
+
+      const network = this.network;
+      const height = this.chain.height;
+      const nameHash = rules.hashName(name);
+      const reserved = rules.isReserved(nameHash, height + 1, network);
+      const [start, week] = rules.getRollout(nameHash, network);
+      const ns = await this.chain.db.getNameState(nameHash);
+
+      let info = null;
+
+      if (ns) {
+        if (!ns.isExpired(height, network))
+          info = ns.getJSON(height, network);
+      }
+
+      return res.json(200, {
+        start: {
+          reserved: reserved,
+          week: week,
+          start: start
+        },
+        info
+      });
+    });
+
+    this.get('/name/hash/:hash', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const hash = valid.bhash('hash');
+
+      if (!hash)
+        throw new Error('Invalid hash.');
+
+      const ns = await this.chain.db.getNameState(hash);
+
+      if (!ns)
+        return res.json(404);
+
+      return res.json(200, { name: ns.name.toString('binary') });
+    });
+
+    this.get('/resource/name/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      if (!name || !rules.verifyName(name))
+        throw new Error('Invalid name.');
+
+      const nameHash = rules.hashName(name);
+      const ns = await this.chain.db.getNameState(nameHash);
+
+      if (!ns || ns.data.length === 0)
+        return null;
+
+      const resource = Resource.decode(ns.data);
+
+      return res.json(200, resource.toJSON());
+    });
+
+    this.get('/proof/name/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      if (!name || !rules.verifyName(name))
+        throw new Error('Invalid name.');
+
+      const hash = this.chain.tip.hash;
+      const height = this.chain.tip.height;
+      const root = this.chain.tip.treeRoot;
+      const key = rules.hashName(name);
+      const proof = await this.chain.db.prove(root, key);
+
+      return res.json(200, {
+        hash: hash.toString('hex'),
```

Comment:
Its not particularly clear what `hash` is in this context, what about `tip`? It is used this way for node `GET /`
https://github.com/handshake-org/hsd/blob/41ff9172ba32ce4b2af78dc11ec4045e0b37168e/lib/node/http.js#L126

I think `key` should be renamed to `nameHash`